drivers/spi/spi_nrfx_spim.c

```diff
@@ -57,6 +57,39 @@ struct spi_nrfx_config {
 
 static void event_handler(const nrfx_spim_evt_t *p_event, void *p_context);
 
+static inline uint32_t get_nrf_spim_frequency(uint32_t frequency)
+{
+	/* Get the highest supported frequency not exceeding the requested one.
+	 */
+	if (frequency < KHZ(250U)) {
+		return KHZ(125U);
```

Comment:
Could we take this opportunity and rephrase this function a bit? I think the following approach could make it clearer:
```
	if (NRF_SPIM_HAS_32_MHZ_FREQ && frequency >= MHZ(32)) {
		return MHZ(32);
	} else if (NRF_SPIM_HAS_16_MHZ_FREQ && frequency >= MHZ(16)) {
		return MHZ(16);
	} else if (frequency >= MHZ(8)) {
		return MHZ(8);

	...

	} else {
		return KHZ(125);
	}
```